explorer/src/components/JupiterSwapModal.tsx

```diff
@@ -0,0 +1,562 @@
+			setFormValue((prevValue) => {
+				return {
+					...prevValue,
+					outputMint: new PublicKey(token.address),
+				};
+			});
+			setWalletBalance(token.address, "receive");
+			setOutputMint(new PublicKey(token.address));
+			setOutputToken(token);
+		}
+		setShowTokenSearch(false);
+	};
+
+	const onTokenSearchChange = (value: string) => {
+		setTokenSearch(value);
+	}
+
+	useEffect(() => {
+		const options = allTokens.filter((token) => {
+			const searchLower = tokenSearch.toLowerCase();
+			return (
+				token.name.toLowerCase().includes(searchLower) ||
+				token.symbol.toLowerCase().includes(searchLower) ||
+				token.address.toLowerCase() === searchLower
+			);
+		})
+		setTokens(options);
+	}, [tokenSearch, allTokens]);
+
+	const onAggregatorSearchChange = (value: string) => {
+		setAggregatorSearch(value);
+	}
+
+	useEffect(() => {
+		const options = aggregators.filter((aggregator) => {
+			const searchLower = aggregatorSearch.toLowerCase();
+			return aggregator.name.toLowerCase().includes(searchLower);
+		})
+		setAggregatorOptions(options);
+	}, [aggregatorSearch]);
+
+	const onClickSlippageOptionBtn = (selectedIdx: number) => {
+		setSelectSlippage(slippageOptions[selectedIdx]);
+	};
+
+	const saveSlippageSettings = () => {
+		if (selectSlippage || inputSlippage.length) {
+			setFormValue((prevValue) => {
+				return {
+					...prevValue,
+					slippage: selectSlippage ?? Number(inputSlippage),
+				};
+			});
+			setShowSlippageSettings(false);
+		}
+	};
+
+	const jupiter = useJupiter({
+		amount: exchangeAmount,
+		inputMint,
+		outputMint,
+		slippage: 1,
+		debounceTime: 250,
+	});
+	const {
+		allTokenMints,
+		routeMap,
+		exchange,
+		refresh,
+		lastRefreshTimestamp,
+		loading,
+		routes,
+		error,
+	} = jupiter;
+	useEffect(() => {
+		if (routes) {
+			setDisplayRoutes(routes?.slice(0, 2));
+			setAdditionalRoutes(routes?.slice(2, routes.length));
+		}
+	}, [routes]);
+	const setTokenSearchValues = (type: any) => {
+		setTokenSearchType(type);
+		setShowTokenSearch(true)
+	}
+	const getTokenSymbol = (tokenAddress: String) => {
+		let symbol = null
+		for (let i = 0; i < allTokens.length; i++) {
+			if (allTokens[i].address === tokenAddress) {
+				symbol = allTokens[i].symbol
+				break;
+			}
+		}
+		return symbol;
+	}
+	const fetchSwapPath = (route: any) => {
+		const path: any[] = [];
+		route.marketInfos.forEach((item: any) => {
+			if (path.length === 0) {
+				const tokenSymbol = getTokenSymbol(item.inputMint.toBase58());
+				path.push(tokenSymbol);
+			}
+			if (path.length > 0 && path[path.length - 1] !== getTokenSymbol(item.inputMint.toBase58())) {
+				path.push(getTokenSymbol(item.inputMint.toBase58()));
+			}
+			path.push(getTokenSymbol(item.outputMint.toBase58()));
+		});
+		return path;
+	}
+	const swapUserInputTokens = () => {
+		const swapInputToken = inputToken;
+		const swapOutputToken = outputToken;
+		setTokenValues(swapOutputToken, "pay");
+		setTokenValues(swapInputToken, "receive");
+	}
+
```

Comment:
- Need clarification on the functionality of this function, but some thoughts immediately when seeing this is that, when watching variables that are part of useState, good to use useMemo or useEffect to ensure that the latest variable is being read.
- Also as mentioned somewhere above, please refrain from naming your custom function as set[SomeName], as it is misleading and might conflict with the setState functions for the various useStates in this file.
- Also, it appears that this function is setting the state of another variable when the inputToken and outputToken variable changes, perhaps a useEffect might be more suitable for this functionality? (need clarification on the functionality of this function)